src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
'keep state' comment was to explain why `startProcessing` have document and mode (in future) parameters. That was not proposal for such redesign.

I'm not sure that understand why batch processing need "start" & "end" methods? I see only requirements to call `flush` method after processing done.